src/document.c

```diff
@@ -396,7 +396,7 @@ FIELD_PREPROCESSOR(fulltextPreprocessor) {
   size_t fl;
   const char *c = RedisModule_StringPtrLen(field->text, &fl);
   if (FieldSpec_IsSortable(fs)) {
-    RSSortingVector_Put(aCtx->sv, fs->sortIdx, (void *)c, RS_SORTABLE_STR);
+    RSSortingVector_Put(aCtx->sv, fs->sortIdx, (void *)c, RS_SORTABLE_STR, fs->options & FieldSpec_UNF);
```

Comment:
I like to booleanize bitmask expresions (`!!(fs->options & FieldSpec_UNF)`). Not a must.